src/routes/v2/sgidAuth.ts

```diff
@@ -86,25 +90,130 @@ export class SgidAuthRouter {
 
     const { nonce, codeVerifier } = cookieData
 
-    // Exchange the authorization code and code verifier for the access token, then use the access token to retrieve user's email
-    await this.sgidAuthService
+    // Exchange the authorization code and code verifier for the access token, then use the access token to retrieve user's data
+    const userDataRes = await this.sgidAuthService
       .retrieveSgidAccessToken({ authCode, nonce, codeVerifier })
       .andThen(({ accessToken, sub }) =>
         // We can immediately process the access token - we only need to retrieve the email from sgid
-        this.sgidAuthService.retrieveSgidUserEmail(accessToken, sub)
+        this.sgidAuthService.retrieveSgidUserEmails(accessToken, sub)
       )
-      .andThen((email) => {
-        if (!email || !email.endsWith("@open.gov.sg")) {
+      .andThen((officerDetails) => {
```

Comment:
Super nit: is it actually officer emails or officer details? If it's details then we should update the prior method call, otherwise this var name

Reply:
officer details! https://github.com/isomerpages/isomercms-backend/pull/976/commits/ee9bbe1313676bbf9580e93594a03fd6d302bc00